scheduler/src/cook/mesos/api.clj

```diff
@@ -932,6 +941,24 @@
       (log/debug e "Unable to retrieve directory path for" task-id "on agent" agent-hostname)
       nil)))
 
+(defn fetch-compute-cluster-map
+  "Converts a compute cluster entity as a map representing the fields. For legacy instances
+  that predate cook-cluster, inline the default cluster, pulled from config."
+  [db entity]
+  (if entity
+    {:compute-cluster-type
+     (case (:compute-cluster/type entity)
+       (:compute-cluster.type/mesos) :compute-cluster-type-mesos
+       :compute-cluster-type-unknown)
+     :compute-cluster-name (:compute-cluster/cluster-name entity)
+     :mesos-framework-id (:compute-cluster/mesos-framework-id entity)}
+    (assoc
```

Comment:
Another option here is instead of having this in code, running a backfill script. In any case I think I'd rather return `nil` or do a manual backfill than have this in code.